feat(wallet)_: coingecko improvements and aave tokens list added

saledjenic opened the pull request:
The work in this PR was initiated by this issue https://github.com/status-im/status-desktop/issues/17377 but since we faced some obstacles with the approach to combine all tokens by symbol, the work done here is the first step towards fetching the tokens list dynamically and splitting everything and maintaining tokens per chain+contract, cause that's the unique identifier for tokens. 

The work is so far split into 3 commits.
- The first one is about making the code around coingecko requests cleaner
- The second one adds requests for fetching coingecko platforms (supported chains) we will need this when implementing dynamic tokens update
- The third is adding aave tokens list to the static tokens store. Within this one, the code for downloading the uniswap tokens list is updated so that it downloads both (uniswap and aave) tokens lists and prepares tokens in a certain form which are later added to a single combined tokens store.

saledjenic committed:
chore(wallet)_: coingecko client code reoorganization

saledjenic committed:
feat(wallet)_: coingecko platform request added

Necessary for fetching tokens dynamically.

saledjenic committed:
feat(wallet)_: aave tokens list added to the static tokens store

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (77)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 4196c273 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/1/) | 2025-02-25 17:05:23 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250225-170238-4196c2-pr6363.zip) |
| :heavy_check_mark: | 4196c273 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/1/) | 2025-02-25 17:05:33 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250225-170236-4196c2-pr6363.aar) |
| :heavy_check_mark: | 4196c273 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/1/) | 2025-02-25 17:06:35 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250225-170239-4196c2-pr6363.zip) |
| :heavy_check_mark: | 4196c273 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/1/) | 2025-02-25 17:07:58 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250225-170238-4196c2-pr6363.zip) |
| :heavy_check_mark: | 4196c273 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/1/) | 2025-02-25 17:08:12 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250225-170245-4196c2-pr6363.zip) |
| :heavy_check_mark: | 4196c273 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/1/) | 2025-02-25 17:08:22 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250225-170241-4196c2-pr6363.zip) |
| :heavy_multiplication_x: | 4196c273 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/1/) | 2025-02-25 17:15:53 | ~13 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/1//consoleText) |
| :heavy_multiplication_x: | 4196c273 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/1/) | 2025-02-25 17:34:55 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 7fd368db | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/2/) | 2025-02-26 10:16:29 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250226-101323-7fd368-pr6363.aar) |
| :heavy_check_mark: | 7fd368db | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/2/) | 2025-02-26 10:16:35 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250226-101324-7fd368-pr6363.zip) |
| :heavy_check_mark: | 7fd368db | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/2/) | 2025-02-26 10:17:41 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250226-101331-7fd368-pr6363.zip) |
| :heavy_check_mark: | 7fd368db | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/2/) | 2025-02-26 10:18:05 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250226-101325-7fd368-pr6363.zip) |
| :heavy_check_mark: | 7fd368db | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/2/) | 2025-02-26 10:18:52 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250226-101328-7fd368-pr6363.zip) |
| :heavy_check_mark: | 7fd368db | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/2/) | 2025-02-26 10:19:31 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250226-101325-7fd368-pr6363.zip) |
| :heavy_multiplication_x: | 7fd368db | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/2/) | 2025-02-26 10:26:16 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/2//consoleText) |
| :heavy_multiplication_x: | 7fd368db | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/2/) | 2025-02-26 10:44:12 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/2//consoleText) |
| :heavy_multiplication_x: | 7fd368db | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/3/) | 2025-02-26 15:22:47 | ~13 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/3//consoleText) |
| :heavy_multiplication_x: | 7fd368db | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/3/) | 2025-02-26 15:40:48 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 943dea93 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/3/) | 2025-02-27 07:44:59 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250227-074155-943dea-pr6363.aar) |
| :heavy_check_mark: | 943dea93 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/3/) | 2025-02-27 07:45:40 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250227-074156-943dea-pr6363.zip) |
| :heavy_check_mark: | 943dea93 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/3/) | 2025-02-27 07:47:11 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250227-074159-943dea-pr6363.zip) |
| :heavy_check_mark: | 943dea93 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/3/) | 2025-02-27 07:47:34 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250227-074159-943dea-pr6363.zip) |
| :heavy_check_mark: | 943dea93 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/3/) | 2025-02-27 07:47:40 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250227-074203-943dea-pr6363.zip) |
| :heavy_check_mark: | 943dea93 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/3/) | 2025-02-27 07:48:00 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250227-074157-943dea-pr6363.zip) |
| :heavy_multiplication_x: | 943dea93 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/4/) | 2025-02-27 07:54:32 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/4//consoleText) |
| :heavy_multiplication_x: | 943dea93 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/4/) | 2025-02-27 08:13:42 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | d116a9d1 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/4/) | 2025-02-27 08:54:10 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250227-085102-d116a9-pr6363.aar) |
| :heavy_check_mark: | d116a9d1 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/4/) | 2025-02-27 08:54:10 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250227-085103-d116a9-pr6363.zip) |
| :heavy_check_mark: | d116a9d1 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/4/) | 2025-02-27 08:55:18 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250227-085111-d116a9-pr6363.zip) |
| :heavy_check_mark: | d116a9d1 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/4/) | 2025-02-27 08:56:16 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250227-085105-d116a9-pr6363.zip) |
| :heavy_check_mark: | d116a9d1 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/4/) | 2025-02-27 08:56:33 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250227-085107-d116a9-pr6363.zip) |
| :heavy_check_mark: | d116a9d1 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/4/) | 2025-02-27 08:57:33 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250227-085103-d116a9-pr6363.zip) |
| :heavy_multiplication_x: | d116a9d1 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/5/) | 2025-02-27 09:03:28 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/5//consoleText) |
| :heavy_multiplication_x: | d116a9d1 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/5/) | 2025-02-27 09:21:06 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | fa350d61 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/5/) | 2025-02-27 08:57:17 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250227-085415-fa350d-pr6363.aar) |
| :heavy_check_mark: | fa350d61 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/5/) | 2025-02-27 08:57:58 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250227-085416-fa350d-pr6363.zip) |
| :heavy_check_mark: | fa350d61 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/5/) | 2025-02-27 08:59:17 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250227-085521-fa350d-pr6363.zip) |
| :heavy_check_mark: | fa350d61 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/5/) | 2025-02-27 09:02:05 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250227-085637-fa350d-pr6363.zip) |
| :heavy_check_mark: | fa350d61 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/5/) | 2025-02-27 09:03:12 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250227-085737-fa350d-pr6363.zip) |
| :heavy_check_mark: | fa350d61 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/5/) | 2025-02-27 09:04:22 | ~7 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250227-085622-fa350d-pr6363.zip) |
| :heavy_multiplication_x: | fa350d61 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/6/) | 2025-02-27 09:17:22 | ~13 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/6//consoleText) |
| :heavy_multiplication_x: | fa350d61 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/6/) | 2025-02-27 09:52:46 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/6//consoleText) |
| :heavy_multiplication_x: | fa350d61 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/7/) | 2025-02-27 10:14:09 | ~13 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/7//consoleText) |
| :heavy_multiplication_x: | fa350d61 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/7/) | 2025-02-27 10:29:41 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | fe834738 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/6/) | 2025-02-27 13:56:28 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250227-135322-fe8347-pr6363.aar) |
| :heavy_check_mark: | fe834738 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/6/) | 2025-02-27 13:56:32 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250227-135323-fe8347-pr6363.zip) |
| :heavy_check_mark: | fe834738 | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/6/) | 2025-02-27 13:57:26 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250227-135330-fe8347-pr6363.zip) |
| :heavy_check_mark: | fe834738 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/6/) | 2025-02-27 13:58:07 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250227-135326-fe8347-pr6363.zip) |
| :heavy_check_mark: | fe834738 | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/6/) | 2025-02-27 13:58:46 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250227-135326-fe8347-pr6363.zip) |
| :heavy_check_mark: | fe834738 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/6/) | 2025-02-27 13:59:36 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250227-135324-fe8347-pr6363.zip) |
| :heavy_multiplication_x: | fe834738 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/8/) | 2025-02-27 14:06:18 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/8//consoleText) |
| :heavy_multiplication_x: | fe834738 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/8/) | 2025-02-27 14:24:51 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 820c95a4 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/7/) | 2025-03-03 07:19:42 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250303-071551-820c95-pr6363.aar) |
| :heavy_check_mark: | 820c95a4 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/7/) | 2025-03-03 07:19:52 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250303-071553-820c95-pr6363.zip) |
| :heavy_check_mark: | 820c95a4 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/7/) | 2025-03-03 07:20:16 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250303-071552-820c95-pr6363.zip) |
| :heavy_check_mark: | 820c95a4 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/7/) | 2025-03-03 07:20:16 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250303-071559-820c95-pr6363.zip) |
| :heavy_check_mark: | 820c95a4 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/7/) | 2025-03-03 07:21:22 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250303-071555-820c95-pr6363.zip) |
| :heavy_check_mark: | 820c95a4 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/7/) | 2025-03-03 07:21:55 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250303-071552-820c95-pr6363.zip) |
| :heavy_multiplication_x: | 820c95a4 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/9/) | 2025-03-03 07:28:04 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/9//consoleText) |
| :heavy_check_mark: | 820c95a4 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/9/) | 2025-03-03 07:47:05 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/9//consoleText) |
| :heavy_multiplication_x: | 820c95a4 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/10/) | 2025-03-03 08:45:41 | ~17 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 8f365ba9 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/8/) | 2025-03-03 10:09:14 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250303-100606-8f365b-pr6363.aar) |
| :heavy_check_mark: | 8f365ba9 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/8/) | 2025-03-03 10:09:25 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250303-100607-8f365b-pr6363.zip) |
| :heavy_check_mark: | 8f365ba9 | [#8](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/8/) | 2025-03-03 10:10:11 | ~3 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250303-100616-8f365b-pr6363.zip) |
| :heavy_check_mark: | 8f365ba9 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/8/) | 2025-03-03 10:11:37 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250303-100612-8f365b-pr6363.zip) |
| :heavy_check_mark: | 8f365ba9 | [#8](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/8/) | 2025-03-03 10:12:03 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250303-100610-8f365b-pr6363.zip) |
| :heavy_check_mark: | 8f365ba9 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/8/) | 2025-03-03 10:12:15 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250303-100609-8f365b-pr6363.zip) |
| :heavy_multiplication_x: | 8f365ba9 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/11/) | 2025-03-03 10:18:58 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/11//consoleText) |
| :heavy_check_mark: | 8f365ba9 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/10/) | 2025-03-03 10:36:22 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | bb062161 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/9/) | 2025-03-03 12:23:38 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250303-122033-bb0621-pr6363.aar) |
| :heavy_check_mark: | bb062161 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/9/) | 2025-03-03 12:23:47 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250303-122034-bb0621-pr6363.zip) |
| :heavy_check_mark: | bb062161 | [#9](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/9/) | 2025-03-03 12:24:47 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250303-122041-bb0621-pr6363.zip) |
| :heavy_check_mark: | bb062161 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/9/) | 2025-03-03 12:26:08 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250303-122037-bb0621-pr6363.zip) |
| :heavy_check_mark: | bb062161 | [#9](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/9/) | 2025-03-03 12:26:38 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250303-122037-bb0621-pr6363.zip) |
| :heavy_check_mark: | bb062161 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/9/) | 2025-03-03 12:26:54 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250303-122035-bb0621-pr6363.zip) |
| :heavy_check_mark: | bb062161 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/12/) | 2025-03-03 12:33:17 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/12//consoleText) |
| :heavy_multiplication_x: | bb062161 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/11/) | 2025-03-03 12:52:07 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/11//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | d2292657 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/10/) | 2025-03-03 13:03:13 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250303-130020-d22926-pr6363.zip) |
| :heavy_check_mark: | d2292657 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/10/) | 2025-03-03 13:03:28 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250303-130018-d22926-pr6363.aar) |
| :heavy_check_mark: | d2292657 | [#10](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/10/) | 2025-03-03 13:04:30 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250303-130028-d22926-pr6363.zip) |
| :heavy_check_mark: | d2292657 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/10/) | 2025-03-03 13:04:45 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250303-130022-d22926-pr6363.zip) |
| :heavy_check_mark: | d2292657 | [#10](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/10/) | 2025-03-03 13:06:21 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250303-130023-d22926-pr6363.zip) |
| :heavy_check_mark: | d2292657 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/10/) | 2025-03-03 13:06:23 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250303-130021-d22926-pr6363.zip) |
| :heavy_check_mark: | d2292657 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/13/) | 2025-03-03 13:12:06 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/13//consoleText) |
| :heavy_check_mark: | d2292657 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/12/) | 2025-03-03 13:30:49 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/12//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 34177ceb | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6363/11/) | 2025-03-05 14:15:33 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250305-141228-34177c-pr6363.aar) |
| :heavy_check_mark: | 34177ceb | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6363/11/) | 2025-03-05 14:15:43 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250305-141229-34177c-pr6363.zip) |
| :heavy_check_mark: | 34177ceb | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6363/11/) | 2025-03-05 14:16:30 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250305-141232-34177c-pr6363.zip) |
| :heavy_check_mark: | 34177ceb | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6363/11/) | 2025-03-05 14:17:59 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250305-141230-34177c-pr6363.zip) |
| :heavy_check_mark: | 34177ceb | [#11](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6363/11/) | 2025-03-05 14:18:03 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250305-141234-34177c-pr6363.zip) |
| :heavy_check_mark: | 34177ceb | [#11](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6363/11/) | 2025-03-05 14:18:13 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250305-141238-34177c-pr6363.zip) |
| :heavy_check_mark: | 34177ceb | [#14](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/14/) | 2025-03-05 14:25:35 | ~12 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6363/14//consoleText) |
| :heavy_check_mark: | 34177ceb | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/13/) | 2025-03-05 14:43:32 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6363/13//consoleText) |

dlipicar commented on services/wallet/service.go:
```diff
@@ -139,7 +139,7 @@ func NewService(
 		User:     config.WalletConfig.StatusProxyMarketUser,
 		Password: config.WalletConfig.StatusProxyMarketPassword,
 	})
-	marketManager := market.NewManager([]thirdparty.MarketDataProvider{cryptoCompare, coingecko, cryptoCompareProxy}, feed)
```
why are we removing direct access to cryptocompare?

alaibe replied:
we should not!

alaibe replied:
but next line make coingecko first which we should not

saledjenic replied:
100% we should not remove it. This was not intentionally removed. I made that change to test some things locally cause otherwise always cryptocompare was used and then forgot to discard that line and it got committed. Reversing it now.

dlipicar commented on services/wallet/thirdparty/market/coingecko/client.go:
```diff
@@ -14,6 +12,8 @@ import (
 	"github.com/status-im/status-go/services/wallet/thirdparty/utils"
 )
 
+var baseURL = "https://api.coingecko.com/api/v3"
```
having this as a member variable instead of a global was better (for example, you could have direct access and proxy-access to coingecko simply by having two instances pointing to different urls)

dlipicar commented on services/wallet/thirdparty/market/coingecko/responses.go:
better add something like `test_data` in the name of these kinds of files

dlipicar commented on services/wallet/token/tokens_sources.go:
```diff
@@ -0,0 +1,29 @@
+	AaveTokenListID: {
+		Name:       "Aave Token List",
+		SourceURL:  "https://tokenlist.aave.eth.link",
+		OutputFile: "services/wallet/token/aave.go",
```
no schema specified here? 🤔

dlipicar commented on services/wallet/token/downloader/main.go:
```diff
@@ -100,18 +132,37 @@ func main() {
+	}
+
+	version, timestamp := bytesToVersionAndTime(body)
+	token.TokensSources[key] = token.TokensSource{
```
hmm editing a global variable from an offline-running script raises a couple red flags in my mind.
- When running this offline we won't have access to these fields, unless there's some magic I'm not aware of
- Whenever we decide to run this in runtime you'll want to make access to that struct thread-safe since the token list is accessed from a bunch of different places and this will probably get updated in response to some time-based event.

saledjenic replied:
That's not what I wanted, opened the PR a bit early.

alaibe commented on services/wallet/thirdparty/market/coingecko/client.go:
```diff
@@ -208,27 +145,16 @@ func (c *Client) FetchPrices(symbols []string, currencies []string) (map[string]
-
-	url := fmt.Sprintf("%s/simple/price", c.baseURL)
-	response, err := c.httpClient.DoGetRequest(context.Background(), url, params, nil)
+	simplePrices, err := c.FetchSmplePrice(context.Background(), maps.Values(ids), currencies)
```
typo

saledjenic commented:
@alaibe @dlipicar all comments addressed. Thanks.

dlipicar commented on services/wallet/token/token.go:
```diff
@@ -162,6 +165,44 @@ func prepareTokens(networkManager network.ManagerInterface, stores []Store) []*T
+
+	parentFolder := filepath.Dir(filename)
+
+	file, err := os.Open(fmt.Sprintf("%s/../../../%s", parentFolder, TokenStoresDetailsFile))
```
I don't think this file will be there in packaged installations. Also, picking stuff up from some plain text file in the filesystem at runtime feels kind of insecure (something like a bad actor modifies this file to make it seem like a shady token is legit)

saledjenic replied:
@dlipicar do you have a better idea?

saledjenic replied:
If we keep that in the database or in any file in which location will be sent as a part of the client's config, we will need to be very careful about the path sent from the client and the path used when downloading new tokens lists.

This way we're sharing the same constant (path location) and will work transparently for both clients (mobile and desktop).

saledjenic replied:
But I believe that after making the tokens lists auto-refreshing (on the app start + every 30 mins) we can remove all this.

dlipicar replied:
Yeah I feel the DB is a more proper spot to store all this. You could have tables `token_list_info` and `token_lists`.

This would only be for the runtime-updating list of tokens, we'd have the token list version we ship along with our status-go binaries to start with (in case the download of a newer version fails for whatever reason).

We're using a similar "built-in + runtime" combination for networks, where we have the "default" configuration but we let the user modify it, so we store that data in the db as well.

saledjenic replied:
@dlipicar all this is cause we made a change on the fly. Requirements for this PR have been changed a few times this week and based on the last conclusion all this will be changed soon, cause we will auto-refresh tokens.

dlipicar commented on services/wallet/token/token.go:
```diff
@@ -174,8 +215,12 @@ func NewTokenManager(
 	accountsDB *accounts.Database,
 	tokenBalancesStorage TokenBalancesStorage,
 ) *Manager {
+	err := loadTokensStoreDetails()
```
accessing the filesystem in the constructor is a no-go for me, you also have no way of reacting to errors. Better leave all these heavy operations for `Start()`

saledjenic replied:
@dlipicar I know that, the only reason why I consciously add it here is to have `TokensSources` ready before creating stores in case somewhere we use those details but they are not ready yet.

saledjenic replied:
Also as mentioned in [this](https://github.com/status-im/status-go/pull/6363#discussion_r1972413393) comment all that will be changed soon, cause we will have auto-refresh tokens and for that PR we will keep fetched token details in the db. 

We started with this like building the auto-refresh feature, but then gave up on that and decided just to add aave list. But today concluded to have the auto-refresh feature.

yakimant reviewed (approved):
Makefile looks good =D

alaibe reviewed (commented):
did u use https://tokenlists.org/token-list?url=https://raw.githubusercontent.com/bgd-labs/aave-address-book/main/tokenlist.json
this aave token list, I cannot see wsteth in the new token but is inside the aave token list

saledjenic commented:
@alaibe no, I did use this one https://tokenlist.aave.eth.link 
I've just updated it to https://raw.githubusercontent.com/bgd-labs/aave-address-book/main/tokenlist.json and `wstETH` is there now.

saledjenic committed:
fix(wallet)_: fixed difference in handling market data

Cryptocompare client was returning 0 values for tokens that are not supported by the source,
while Coingecko was not returning any mentions of such tokens in the response. Now updated
to match the handling we have for cryptocompare.

This fix also fixes the functional tests flakyness, cause if the fetching market values from the
primary option (cryptocompare) fails, fallback to coingecko (due to different handling) will also
fail in 100% cases.

saledjenic commented:
I've just added one more commit to this PR, which aligns cryptocompare and coingecko marked data handling and fixes functional tests.

@Cuteivist could you have a look at this commit? I see you've made some changes there in the past.
- https://github.com/status-im/status-go/pull/6363/commits/bb062161ac1332b4d40737f762b5b1c888ac97ac

saledjenic committed:
chore(wallet)_: remove unused field from the token type

Removed fields:
- `TokenListID`

saledjenic committed:
chore(wallet)_: associating the token list timestamp with the corresponding field in the response

Before this change updatedAt was pointing to the when the last list from the stores was updated. That's sorted out.

`LastUpdateTimestamp` field added to the `List` type. A client needs to display the time of the last update.

friofry commented on services/wallet/token/aave.go:
```diff
@@ -0,0 +1,5737 @@
+package token
```
I guess the `aaveTokens` is automatically generated
It would be great to have a script that prepares(or downloads) this static data into a Json file
and embed it with something like this:
```
//go:embed data/resource.json
var resource []byte
```
(as `make generate` processes mock :) )

saledjenic replied:
@friofry yes, we already have something like that, have a look at `services/wallet/token/downloader/main.go` file. That one we run when we want to update the token lists. But we're about to kick off with the development of auto-refresh token lists.
